Mark-WinScreen.

MV1090 opened the pull request:
I changed the gameOverMenu to endGameMenu, and added two text boxes which display the appropriate text at the and of the game, if all enemy's are dead the win text displays, if all lives are lost the game over text displays. The win situation is stored in a function and called from the GameManager, this hasn't been tested yet. I also added a line to the start button to call reset() from the GameManager, because if the player lost or won, then went to the main menu, pressed the start button again to restart the game, the game didn't reset, it just displayed the previous game's end state. If you want to do this a different way let me know, I mainly added it to test what I had done, and for this it works fine.

Commit:
Mark-WinScreen.

I changed the gameOverMenu to endGameMenu, and added two text boxes which display the appropriate text at the and of the game, if all enemys are dead the win text displays, if all lives are lost the game over text displays. The win situation is stored in a function and called from the gameManager, this hasn't been tested yet.
I also added a line to the start button to call reset() from the gameManager, because if the player lost or won, went to the main menu, pressed the start button, the game didn't reset, it just displayed the previous game. If you want to do this a different way let me know, I mainly added it to test what I had done.

SolomanDavis commented on Assets/_Scripts/Managers/CanvasManager.cs:
```diff
@@ -117,9 +124,19 @@ protected override void Update()
 
         if (GameManager.Instance.Lives == 0)
         {
-            GameOverMenu.SetActive(true);
+            EndGameMenu.SetActive(true);
+            WinText.enabled = false;
+            GameOverText.enabled = true;
         }
```
Galaga actually treats the number of `Lives` as the number of ships that the player has left to use, including the one the player is actually using.

![image](https://github.com/SolomanDavis/VGP145-S24-capstone-retro/assets/16367411/28c58cc2-8244-43c8-8c81-982e1de2f348)

You can see from this screenshot^ that it's possible to have `Lives == 0` but still be in play. It's only when you die while you have 0 lives left that the game is actually over, not when your lives drops from 1 -> 0.

This is accounted for in the `GameManager` when setting `Lives`: https://github.com/SolomanDavis/VGP145-S24-capstone-retro/blob/4d2104b110ff82e969362e843c8e2249964c9fa8/Assets/_Scripts/Managers/GameManager.cs#L23-L38

when setting, if the new value for `Lives` is less than 0, then we call `GameManager::GameOver()` https://github.com/SolomanDavis/VGP145-S24-capstone-retro/blob/4d2104b110ff82e969362e843c8e2249964c9fa8/Assets/_Scripts/Managers/GameManager.cs#L55-L60 which doesn't do anything yet. I think this method should be implemented to call a `CanvasManager::GameOver()` similar to `CanvasManager::GameWon()` that you already have.

It's also nice because it means the `CanvasManager` doesn't need to check the number of lives on every update.

MV1090 replied:
Changes made.

SolomanDavis reviewed (changes requested):
All looks great to me and I really like the `EndGameMenu` idea since it is indeed very reusable 🎉 

Just had one comment though around how the `GameOver` is triggered. Currently through an `Update()` check but recommend to be done via a `CanvasManager::GameOver()` function call due to how `Lives` behaves in this game.

Commit:
Merge branch 'master' of github.com:SolomanDavis/VGP145-S24-capstone-retro into Mark-WinScreen

Commit:
Mark-WinScreen

Game over text is displayed on the screen when GameOver method is called, I set the Method in the GameManager GameOver Method.

SolomanDavis reviewed (approved):
Awesome - LGTM
